### PR TITLE
Info-window pops up with dummy information when clicked on a building

### PR DESCRIPTION
**Pull request by bogalla**
when you click on a building, an info window pops up with information about the building. The three buildings with real info are Hall, LB and EV.
lbinfo

### CODE REVIEW

File: app/src/main/java/com/example/campusguide/Bootstrapper.kt [hunk 1]
```diff
@@ -33,6 +34,8 @@ class Bootstrapper constructor(activity: MapsActivity) {
         activity.setOnSearchClickedListener(search)
         activity.addActivityResultListener(search)
 
+        //building info wimdow
+
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/Bootstrapper.kt]: Was there something that was supposed to be here? If not, delete

File: app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import com.google.android.gms.maps.model.Polygon
+
+class BuildingClickListener(private val context: Context, private val googleMap: GoogleMap) : GoogleMap.OnPolygonClickListener{
+
+    fun buildingInfoWindow(location: LatLng, info: InfoWindowData) {
+        val customInfoWindow = CustomInfoWindow(context)
+        googleMap.setInfoWindowAdapter(customInfoWindow)
+
+        // Clears any existing markers from the GoogleMap
+        //googleMap.clear()
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt]: Shouldn't this be kept? If not, delete

File: app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        val markerOptions = MarkerOptions()
+
+        // Setting position on the MarkerOptions
+        markerOptions.position(location)
+
+        //Set Custom InfoWindow Adapter
+        googleMap.setInfoWindowAdapter(customInfoWindow)
+
+        // Animating to the currently touched position
+        googleMap.animateCamera(CameraUpdateFactory.newLatLng(location))
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt]: This makes the app center on the marker, cutting off the top half of the window.  Is there any way to center the camera on the widow instead? It should be a constant offset

File: app/src/main/java/com/example/campusguide/MapsActivity.kt [hunk 4]
```diff
@@ -70,4 +70,7 @@ class MapsActivity : AppCompatActivity() {
         val navigateButton: FloatingActionButton = findViewById(R.id.navigateButton)
         navigateButton.setOnClickListener(listener)
     }
-}
\ No newline at end of file
+
+
+
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/MapsActivity.kt]: Ahh so much whitespace

File: app/src/main/java/com/example/campusguide/map/infoWindow/CustomInfoWindow.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        val services = view.findViewById<TextView>(R.id.services)
+        val events = view.findViewById<TextView>(R.id.events)
+
+        val infoWindowData = marker.tag as InfoWindowData?
+
+        symbol.setText(infoWindowData?.symbol)
+        full_name.setText(infoWindowData?.full_name)
+        address.setText(infoWindowData?.address)
+        services.setText(infoWindowData?.services)
+        events.setText(infoWindowData?.events)
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/map/infoWindow/CustomInfoWindow.kt]: use property access syntax? ex: `symbol.text = infoWindowData?.symbol`

File: app/src/main/res/layout/custom_info_marker.xml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        android:id="@+id/symbol"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textAppearance="@style/TextAppearance.AppCompat.Headline"
+        android:textSize="20sp"
+        android:layout_marginTop="10dp"
+        android:layout_marginBottom="10dp"
+        android:textColor="@color/colorAccent" />
+    <TextView
+        android:id="@+id/full_name"
```
Comment (jun-loke) [inline app/src/main/res/layout/custom_info_marker.xml]: Decide if you're doing snake case or camel case (see comment line 49)

File: app/src/main/res/layout/custom_info_marker.xml [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+        android:id="@+id/events"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textSize="20sp"
+        android:layout_marginTop="10dp"
+        android:layout_marginBottom="10dp"
+        android:textColor="@color/colorAccent"/>
+
+    <Button
+        android:id="@+id/btnAddARoom"
```
Comment (jun-loke) [inline app/src/main/res/layout/custom_info_marker.xml:48]: Decide if you're doing snake case or camel case (see comment line 16)

File: app/src/main/java/com/example/campusguide/map/infoWindow/CustomInfoWindow.kt [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.Marker
+
+import com.example.campusguide.R
+import com.example.campusguide.map.infoWindow.InfoWindowData
+import com.google.android.gms.maps.CameraUpdateFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.MarkerOptions
+import com.google.android.gms.maps.model.Polygon
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/map/infoWindow/CustomInfoWindow.kt]: Imports not being used. Delete

**Review by jun-loke — CHANGES_REQUESTED** [review]
Just a couple clean up things. But otherwise seems good.  
Also, what's the deal with the buildship stuff? should that be gitignored?

File: app/src/main/java/com/example/campusguide/map/infoWindow/CustomInfoWindow.kt [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+        val services = view.findViewById<TextView>(R.id.services)
+        val events = view.findViewById<TextView>(R.id.events)
+
+        val infoWindowData = marker.tag as InfoWindowData?
+
+        symbol.text = infoWindowData?.symbol
+        fullName.text = infoWindowData?.fullName
+        address.text = infoWindowData?.address
+        services.text = infoWindowData?.services
+        events.text = infoWindowData?.events
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/map/infoWindow/CustomInfoWindow.kt]: Does this need to be assigned to variables? Can you just do something like 
`view.findViewById<TextView>(R.id,symbol).text = infoWindowData?.symbol` 
for all of them?

File: app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    private fun getPolygonCenterPoint(polygonPointsList: ArrayList<LatLng>): LatLng? {
+        var centerLatLng: LatLng? = null
+        val builder: LatLngBounds.Builder = LatLngBounds.Builder()
+        for (i in 0 until polygonPointsList.size) {
+            builder.include(polygonPointsList[i])
+        }
+        val bounds: LatLngBounds = builder.build()
+        centerLatLng = bounds.center
+        return centerLatLng
+    }
```
Comment (jun-loke) [inline app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt:54]: Could be worth exporting or making this a attribute? Indoor floor plans also needs the center of buildings

Comment (bogalla) [reply app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt:54]: exporting in like a static class or something you mean?

File: app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt (hunk 2 above)
Comment (bogalla) [reply app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt]: fixed

File: app/src/main/java/com/example/campusguide/map/infoWindow/CustomInfoWindow.kt (hunk 9 above)
Comment (bogalla) [reply app/src/main/java/com/example/campusguide/map/infoWindow/CustomInfoWindow.kt]: true dat! done

File: app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt (hunk 10 above)
Comment (jun-loke) [reply app/src/main/java/com/example/campusguide/map/infoWindow/BuildingClickListener.kt:54]: maybe. But we'll discuss that when we get to it

**Review by jun-loke — APPROVED** [review]
Good to go!

**Review by jun-loke — APPROVED** [review]
Looks good! I think the padding around the text could be a bit bigger. But that can be changed at a different time
image

**Pull request merged**